samples/net/mqtt/doc/architecture.rst

```diff
@@ -0,0 +1,176 @@
+| Trigger     | None             |                       | Sends messages on the trigger channel at an interval set by the :kconfig:option:`CONFIG_MQTT_SAMPLE_TRIGGER_TIMEOUT_SECONDS` and upon a button press.                   |
++-------------+------------------+-----------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| Sampler     | Trigger          | Subscriber            | Samples data everytime a message is received on the trigger channel.                                                                                                    |
+|             |                  |                       | The sampled payload is sent on the payload channel.                                                                                                                     |
++-------------+------------------+-----------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| Transport   | Network          | Subscriber            | Handles MQTT connection. Will auto connect and keep the MQTT connection alive as long as the network is available.                                                      |
+|             | Payload          |                       | Receives network status messages on the network channel. Publishes messages received on the payload channel to a configured MQTT topic.                                 |
++-------------+------------------+-----------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| Network     | None             |                       | Auto connects to either Wi-Fi or LTE after boot, depending on the board and the sample configuration. Sends network status messages on the network channel.             |
++-------------+------------------+-----------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| LED         | Network          | Listener              | Listens to changes in the network status received on the network channel. Displays LED pattern accordingly.                                                             |
+|             |                  |                       | If network is connected, LED 1 on the board will light up. On Thingy:91, the LED turns green                                                                            |
++-------------+------------------+-----------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
```

Comment:
Updated the format of the Kconfig option (application specific Kconfig option). This distorts the table and hence fixes the table alignment.
```suggestion
+-------------+------------------+-----------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Module name | Observes channel | Subscriber / Listener | Description                                                                                                                                                                             |
+=============+==================+=======================+=========================================================================================================================================================================================+
| Trigger     | None             |                       | Sends messages on the trigger channel at an interval set by the :ref:`CONFIG_MQTT_SAMPLE_TRIGGER_TIMEOUT_SECONDS <CONFIG_MQTT_SAMPLE_TRIGGER_TIMEOUT_SECONDS>` and upon a button press. |
+-------------+------------------+-----------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Sampler     | Trigger          | Subscriber            | Samples data everytime a message is received on the trigger channel.                                                                                                                    |
|             |                  |                       | The sampled payload is sent on the payload channel.                                                                                                                                     |
+-------------+------------------+-----------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Transport   | Network          | Subscriber            | Handles MQTT connection. Will auto connect and keep the MQTT connection alive as long as the network is available.                                                                      |
|             | Payload          |                       | Receives network status messages on the network channel. Publishes messages received on the payload channel to a configured MQTT topic.                                                 |
+-------------+------------------+-----------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Network     | None             |                       | Auto connects to either Wi-Fi or LTE after boot, depending on the board and the sample configuration. Sends network status messages on the network channel.                             |
+-------------+------------------+-----------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| LED         | Network          | Listener              | Listens to changes in the network status received on the network channel. Displays LED pattern accordingly.                                                                             |
|             |                  |                       | If network is connected, LED 1 on the board will light up. On Thingy:91, the LED turns green                                                                                            |
+-------------+------------------+-----------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
```